python/pyspark/ml/param/__init__.py

```diff
@@ -40,6 +40,15 @@ def __init__(self, parent, name, doc, expectedType=None):
         self.doc = str(doc)
         self.expectedType = expectedType
 
+    def _copy_new_parent(self, parent):
```

Comment:
I hope this can be implemented in class Params instead of Param.  In Params, if we can get a list of all class attributes of type Param, then we can call this method on them (as is currently done manually by each init method).